### DISCUSSION TITLE
`RunQueryDsl` blanket implementation

### DISCUSSION THREAD

**Discussion by Turbo87** [weiznich/diesel_async#200] [General]
https://github.com/weiznich/diesel_async/blob/v0.5.1/src/run_query_dsl/mod.rs#L644 currently implements the `RunQueryDsl` trait for every existing type in the application, while `diesel` itself [restricts](https://github.com/diesel-rs/diesel/blob/v2.2.5/diesel/src/query_dsl/mod.rs#L1785-L1791) it to `where T: Table`.

Is this an intentional choice?

I just ran into an issue where there was a conflict with the `first()` fn of a slice and the `LimitDsl` due to this blanket implementation.

**Comment by weiznich** [comment]
Thanks for raising this. I think that might be related to #114 

There are several impls for different specific types  in diesel, not only that generic one for `T: Table`. I think I just went with a wild card implementation in the beginning to have something and forgot about that later on. Now if I think about that again it's likely not that good to have that generic impl there, but I do not see an easy way to fix that as some of the types that implement the sync variant in diesel are private types so we cannot just replicate all impls in diesel-async. It's also not possible to just say: Implement this for all `T` that also implement the `RunQueryDsl` trait from diesel, as the connection types would be different. 

The other question is why rustc even believes that these methods would be implemented for slices at all, as the bounds on the method itself exclude that + the arguments are different.
